qcore/helpers.py

```diff
@@ -261,13 +261,23 @@ def copy_public_attrs(source_obj, dest_obj):
 def object_from_string(st):
     """Creates a Python class or function from its fully qualified name.
 
-    :param st: A fully qualified name of a class or a function.
+    :param st: A fully qualified name of a class or a function. In Python 3 this
+        is only allowed to be of text type (unicode). In Python 2, both bytes and unicode
+        are allowed.
     :return: A function or class object.
 
     This method is used by serialization code to create a function or class
     from a fully qualified name.
 
     """
+    if six.PY3:
+        assert isinstance(st, six.text_type), 'only text type is allowed in Python 3'
```

Comment:
Let's raise a TypeError here instead of just failing an assertion. That matches how we handle MarkerObject above.

Also, no need to mention Python 3 here. However, it'd be helpful to include the name of the bad type in the exception message, something like `"st must be str, not %r" % (type(st),)`.

Speaking of which, `st` is a pretty bad parameter name.